fiftyone/server/utils.py

```
@@ -25,43 +25,58 @@ def load_and_cache_dataset(name):
-    _cache[dataset.name] = dataset
+    return _cache_dataset(dataset.name, dataset=dataset)
+
+
+def _cache_dataset(name, dataset=None):
+    _dataset = _cache.get(name)
+
+    if _dataset is not None:
+        try:
+            _dataset.reload()
+            return _dataset
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)